service/resource/deployment/resource.go

```diff
@@ -89,6 +89,13 @@ func New(config Config) (*Resource, error) {
 	if config.TemplateVersion == "" {
 		return nil, microerror.Maskf(invalidConfigError, "config.TemplateURIVersion must not be empty")
 	}
+	// Template version is a full git SHA. It has 40 characters. Whe
+	// testing we can have n/a because it is not set by the CI while
+	// testing locally.
+	if len(config.TemplateVersion) != 40 {
+		config.Logger.Log("warning", fmt.Sprintf("config.TemplateVersion=%#q should be a full git SHA setting to `master`; SHOULD NOT happen in production", config.TemplateVersion))
```

Comment:
Why `%#q`?

Reply:
We often log jsons. I think using `%#q` is a good habit. `fmt.Sprintf("json=%#q", `{"a":"b"}`)` results in `"json=`{"a":"b"}`. No ugly escape `\"`. It makes it a lot easier to read IMO.